src/include/mallocMC/mallocMC_utils.hpp

```diff
@@ -123,18 +132,33 @@ namespace mallocMC
         std::uint32_t lanemask;
         asm("mov.u32 %0, %%lanemask_lt;" : "=r"(lanemask));
         return lanemask;
-#elif defined(__HIP_DEVICE_COMPILE__) && defined(__HIP__)
+#elif(MALLOCMC_DEVICE_COMPILE_PATH && BOOST_COMP_HIP)
         return __lanemask_lt();
 #else
         return 0u;
 #endif
     }
 
+    ALPAKA_FN_ACC inline auto ballot(int pred)
+    {
+#if defined(__CUDA_ARCH__)
+        return __ballot_sync(__activemask(), pred);
+#elif(MALLOCMC_DEVICE_COMPILE_PATH && BOOST_COMP_HIP)
+        // return value is 64bit for HIP-clang
+        return __ballot(pred);
+#else
+        return 1u;
```

Comment:
I think we should `return pred;` here.
On CPU implementations, where there is only 1 thread voting, `ballot()` should return what the 1 thread put in as predicate.

Reply:
ballot is creating a bitmask where the bit of the thread is set where `pred` is non zero. `pred` can be any integer value.

From the [cuda documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-vote-functions)
```
Evaluate predicate for all non-exited threads in mask and return an integer whose Nth bit is set if and only if predicate evaluates to non-zero for the Nth thread
```

Reply:
As soon as alpaka is providing the intrinsic we will remove this code anyway.